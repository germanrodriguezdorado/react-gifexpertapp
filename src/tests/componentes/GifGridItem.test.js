import { shallow } from "enzyme";
import GifGridItem from "../../componentes/GifGridItem";

describe("Pruebas del componente GifGridItem.test.js", () => {

    const titulo = "Algo";
    const url = "http://localhost:4000/img.jpg"

    let wrapper = shallow(<GifGridItem title={titulo} url={url}/>);
    beforeEach(() => {
        wrapper = shallow(<GifGridItem title={titulo} url={url}/>);
    });

    test('Debe desplegar <GifGridItem> correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe tener un pÃ¡rrafo con el title', () => {
        const p = wrapper.find("p");
        expect(p.text().trim()).toBe(titulo);
    });

    test('debe de tener la imagen igual a la url y al src', () => {
        const img = wrapper.find("img");
        expect(img.prop("src")).toBe(url);
        expect(img.prop("alt")).toBe(titulo);
    });

    test('debe de tener la clase "animate__fadeIn"', () => {
        const div = wrapper.find("div");
        expect(div.prop("className")).toContain("animate__fadeIn");
    });
    
    
    
});

