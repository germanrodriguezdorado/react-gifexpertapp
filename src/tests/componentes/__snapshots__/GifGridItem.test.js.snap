// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente GifGridItem.test.js Debe desplegar <GifGridItem> correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Algo"
    src="http://localhost:4000/img.jpg"
  />
  <p>
    Algo
  </p>
</div>
`;
